lib/SPIRV/LLVMToSPIRVDbgTran.cpp

```diff
@@ -1010,9 +1010,16 @@ SPIRVEntry *LLVMToSPIRVDbgTran::transDbgExpression(const DIExpression *Expr) {
   for (unsigned I = 0, N = Expr->getNumElements(); I < N; ++I) {
     using namespace SPIRVDebug::Operand::Operation;
     auto DWARFOpCode = static_cast<dwarf::LocationAtom>(Expr->getElement(I));
+    SPIRVDebug::ExpressionOpCode OC = {};
+
+    // Skip handling of debug intrinsics with unsupported metadata.
+    // TODO: replace this check to ::map call when more DWARF operations are
+    // supported.
+    if (!SPIRV::DbgExpressionOpCodeMap::find(DWARFOpCode, &OC)) {
+      Operations.clear();
```

Comment:
I'm not an expert in debug info processing in the translator and its infrastructure. Yet I have (may be a naive) suggestion not to clear all of the Operations vector in case if a single operation is supported, but instead erasing/not adding this particular operation to it, leaving others to be processed. I can also imagine, that it's a bad suggestion, that can result in a misleading debug information in the module, WDYT?

Reply:
Firstly I also thought that it'd be enough to just skip the unsupported operation and proceed further. 
But there are such DWARF expressions which operands should be processed together, otherwise, they'll make no sense, e.g.:
`!DIExpression(DW_OP_LLVM_arg, 0, DW_OP_LLVM_arg, 1, DW_OP_minus, DW_OP_stack_value)` - if we just ignore `DW_OP_LLVM_arg`, then `0` and `1` values lost their meaning. 

[Language reference ](https://llvm.org/docs/LangRef.html#diexpression) says:
> `DW_OP_LLVM_arg, N` is used in debug intrinsics that refer to more than one value, such as one that calculates the sum of two registers. This is always used in combination with an ordered list of values, such that `DW_OP_LLVM_arg, N` refers to the N``th element in that list.

So, the chosen approach is safer for debug information and that information won't be misleading.

Reply:
I also have no understanding how debugger will interpret such modification of debug info, but my _guess_ is when we remove the expression completely the debugger will something like "I don't know what the value is", and if we remove just one operation from the expression it will give us some value which is not the real/actual/correct. I personally prefer the first option, because the second one is very misleading.